### DISCUSSION TITLE
How can I run a function every time cache is invalidated for a certain endpoint?

### DISCUSSION THREAD

**Discussion by LoyalPotato** [reduxjs/redux-toolkit#2990] [Q&A]
Hey,

So basically, if I have an endpoint with some cached data and after the `keepUnusedDataFor` timer runs out and the cache is cleaned, is there a way for me to "listen" to whenever, before or after, the cache is deleted?

**Comment by markerikson** [comment]
Not directly. What are you trying to accomplish?

**Reply by LoyalPotato** [reply]
I have two data types, let's say Foo and Bar, that have a relation between them. 
I'm caching Foo and what I was looking to do is take advantage of RTKQ's invalidateTags, on the endpoints of Bar, to know when I need to clean the cache of the other data type, Foo, effectively making use of RTKQ's timer for managing how long Foo's data is cached.

**Reply by markerikson** [reply]
Hmm. Maybe using the https://redux-toolkit.js.org/rtk-query/api/createApi#oncacheentryadded lifecycle in that endpoint, and then waiting for the cache entry to be removed?

**Reply by LoyalPotato** [reply]
By waiting you mean using a timer with the same wait time as the `keepUnusedDataFor`? Just to clear any doubt as to what you mean :)

**Reply by markerikson** [reply]
No, I mean `await cacheEntryRemoved`, like in https://redux-toolkit.js.org/rtk-query/usage/streaming-updates#websocket-chat-api .

However, I'm a little confused on what you actually want to accomplish.  You mentioned cache entries being removed, and `invalidateTags`, and these are different tools.  Are you wanting to _refetch_ the other data, or _remove_ it entirely?  We don't currently have an API to flat-out remove a cache entry manually, although it's come up a few times ( #2820 ).

**Reply by LoyalPotato** [reply]
Oh yeah, sorry, I was reading it now and saw what you mentioned. My bad!

**Reply by LoyalPotato** [reply]
I'm sorry for not explaining correctly, but I think what you mentioned might be what I want.
I just want to know when the cache for each endpoint + args is removed so that I can then manage my local cache (not related to RTKQ but to the data stored when making the request using RTKQ).
